### DISCUSSION TITLE
How to use 7.x CSS with cross-platform code?

### DISCUSSION THREAD

**Discussion by Nantris** [mantinedev/mantine#5015] [Q&A]
The requirement to import CSS straight into JavaScript breaks our code that's shared on React Native.

The first thing I tried was a dynamic `import` or `require` but it seems like React Native's Hermes does static analysis so it doesn't matter that I've told it not to load the files - it still complains about them. Specifically it complains with:

> Error: Unable to resolve module @mantine/core/styles.css from ...project/packages/shared/theme-utils/src/ThemeProvider.js: @mantine/core/styles.css could not be found within the project or in these directories: ...

So next I tried to workaround this by importing the CSS straight into our project's SCSS which only gets imported on desktop - but the styles don't seem to be applied with this approach despite there being no error.

Thanks for the great work on `7.x` @rtivital. Can you comment on how are we supposed to handle this?

I tried `@import '@mantine/core/styles.css';` and I get no error, but also no styles. It doesn't appear to be a configuration issue on our end because right below that we have `@import 'font-awesome/css/font-awesome.css';` and it works fine.

So then I tried two other approaches to match the underlying filesystem names: 
- `@import '@mantine/core/esm/index.css';` 
- `@import '@mantine/core/index.css';`
 
 But both yield an error like:

> Error: Package path ./esm/index.css is not exported from package
-----

Font-Awesome's `package.json` has a line like this which might accomodate it:

`"style": "css/font-awesome.css",`

**Comment by Nantris** [comment]
There really seems to be no alternative for us but to import the CSS into our SCSS, and it's baffling to me why this doesn't work. I figured that would be an easy solution but instead nothing happens.

**Comment by rtivital** [comment]
I haven't worked with React Native so I cannot advise much about styles importing in mixed projects. I can propose to only import styles in your web application files.

**Reply by Nantris** [reply]
Thanks for your reply @rtivital. The real core of my question is why importing the Mantine CSS into our SCSS processor doesn't work?

It seems like it's setup _only_ to work as ES6 imports so we can't import it into SCSS, which would fix the problem for us.

I've never experienced this with any other project.

**Comment by Nantris** [comment]
I managed to find a file where we could import these without needing to rebuild our entire tooling or jump through crazy hoops - but I still don't understand why we can't just import the CSS into our SCSS and this may affect others.

**Reply by rtivital** [reply]
I have no idea as well, `@mantine/*` packages provide regular css files in node_modules
